### PR TITLE
Cleanup and delete legacy code

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Cleanup and delete legacy code.

Cleans up MediaExtension and MediaState classes for Edge workflows. Does not add new event listeners to MediaExtension as that will be part of a separate PR.

Deletes legacy code which is not needed or will be rewritten for Edge workflows, including MediaSession (will be rewritten), MediaSessionIDManager, MediaRealTimeService and MediaOfflineService, MediaReporterHelper, MediaDatabase and helper classes. 

Does not delete MediaCollectionTracker, MediaCollectionHitGenerator, MediaHitProcessor, or MediaHit. The hit processor is not needed for Edge workflows, however it will be removed in another PR when the Media Tracker classes are refactored.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/media/src/main/java/com/adobe/marketing/mobile/media/internal/MediaExtension.java [hunk 1]
```diff
@@ -30,11 +33,10 @@ public class MediaExtension extends Extension {
 
     final Map<String, MediaTrackerInterface> trackers;
     MediaState mediaState;
-    MediaRealTimeService mediaRealTimeService;
-    MediaOfflineService mediaOfflineService;
```
Comment (addb) [inline code/media/src/main/java/com/adobe/marketing/mobile/media/internal/MediaExtension.java:36]: Let's add TODO to add/init MediaEventProcessor

File: code/media/src/main/java/com/adobe/marketing/mobile/media/internal/MediaExtension.java [hunk 2]
```diff
@@ -143,30 +135,7 @@ void handleResetIdentities(final Event event) {
...
-                            trackerConfig,
-                            MediaInternalConstants.EventDataKeys.Config.DOWNLOADED_CONTENT,
-                            false);
-        }
-
-        MediaHitProcessor hitProcessor =
-                isDownloadedContent ? mediaOfflineService : mediaRealTimeService;
-        MediaTrackerInterface tracker = new MediaCollectionTracker(hitProcessor, trackerConfig);
-        trackers.put(trackerId, tracker);
+        trackers.clear();
```
Comment (addb) [inline code/media/src/main/java/com/adobe/marketing/mobile/media/internal/MediaExtension.java]: Let's add TODO to notify MediaEventProcessor

File: code/media/src/test/java/com/adobe/marketing/mobile/media/internal/MockMediaOfflineHits.java [hunk 3]
```diff
@@ -155,22 +155,7 @@ class MockMediaOfflineHits {
...
-                        MediaTestConstants.Assurance.SHARED_STATE_NAME, assuranceSharedState);
-            }
-
-            {
-                mediaStateLocHintException = new MediaState();
-                Map<String, Object> identityEventData = new HashMap<>();
-                identityEventData.put(MediaInternalConstants.Identity.LOC_HINT, "exception");
-                mediaStateLocHintException.notifyMobileStateChanges(
-                        MediaInternalConstants.Identity.SHARED_STATE_NAME, identityEventData);
+                mediaState.updateState(configSharedState);
```
Comment (addb) [inline code/media/src/test/java/com/adobe/marketing/mobile/media/internal/MockMediaOfflineHits.java]: We can remove this MockMediaOfflineHits.java file entirely

File: code/media/src/phone/java/com/adobe/marketing/mobile/MediaConstants.java [hunk 4]
```diff
@@ -37,57 +37,57 @@ private StreamType() {}
...
+        public static final String STREAM_FORMAT = "format";
+        public static final String RATING = "rating";
 
         private VideoMetadataKeys() {}
     }
     /** These constant strings define standard player states. */
     public static final class PlayerState {
-        public static final String FULLSCREEN = "fullscreen";
-        public static final String PICTURE_IN_PICTURE = "pictureInPicture";
         public static final String CLOSED_CAPTION = "closedCaptioning";
```
Comment (addb) [inline code/media/src/phone/java/com/adobe/marketing/mobile/MediaConstants.java:86]: closeCaption

Comment (addb) [reply code/media/src/phone/java/com/adobe/marketing/mobile/MediaConstants.java:86]: Tagging this document for reference: https://experienceleague.adobe.com/docs/media-analytics/using/implementation/variables/player-state-parameters.html?lang=en

File: code/media/src/main/java/com/adobe/marketing/mobile/media/internal/MediaExtension.java (hunk 1 above)
Comment (kevinlind) [reply code/media/src/main/java/com/adobe/marketing/mobile/media/internal/MediaExtension.java:36]: I'm adding MediaEventProcessor in the next PR so we don't need the TODO here. I already have the code locally.

File: code/media/src/test/java/com/adobe/marketing/mobile/media/internal/MockMediaOfflineHits.java (hunk 3 above)
Comment (kevinlind) [reply code/media/src/test/java/com/adobe/marketing/mobile/media/internal/MockMediaOfflineHits.java]: This file is already removed.

File: code/media/src/phone/java/com/adobe/marketing/mobile/MediaConstants.java (hunk 4 above)
Comment (kevinlind) [reply code/media/src/phone/java/com/adobe/marketing/mobile/MediaConstants.java:86]: I can update this to be the same as the iOS implementation, but this should be "closedCaptions" or "closedCaptioning".   The "closed" means the captions are not visible to the screen/user.

**Review by addb — APPROVED** [review]

**Pull request merged**